src/tests/Microsoft.PowerFx.Interpreter.Tests/LanguageServiceProtocol/MockSymbolTable.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using Microsoft.PowerFx;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Binding.BindInfo;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+
+/** 
+ * Just a small handy mock of symbol table to be able to customize binder to compute different token types.
+ * Might not be 100% correct but it works and allows testing against different token types.
+*/
+internal class MockSymbolTable : ReadOnlySymbolTable
```

Comment:
I added this in semantic tokens encoder tests file because i didn't intend it to be used outside. I see that you need it outside of that file now. In that case, please add a comment that this should only be used for semantic tokens related tests. This is very specific to that area. People can create new mocks by referencing this but we should just keep it "small handy mock".